.github/workflows/k8s-integration-test.yml

```diff
@@ -0,0 +1,537 @@
+#   ✅ Security scanning (Trivy, Checkov)
+#   ✅ Basic syntax validation for observability and ai-gateway configs
+#
+# Out of Scope (planned for follow-up PRs):
+#   🔄 Observability stack deployment (Prometheus + Grafana)
+#   🔄 AI Gateway end-to-end testing (Envoy Gateway + InferencePool)
+
+on:
+  pull_request:
+    paths:
+      - "deploy/kubernetes/**"
+      - ".github/workflows/k8s-integration-test.yml"
+      - "Dockerfile.extproc"
+      - "tools/kind/**"
+  workflow_dispatch: # Allow manual triggering
+  schedule:
+    # Run nightly at 3:00 AM UTC
+    - cron: "0 3 * * *"
+
+env:
+  KIND_VERSION: v0.20.0
+  KUBECTL_VERSION: v1.28.0
+  KUSTOMIZE_VERSION: v5.2.1
+
+jobs:
+  validate-manifests:
+    name: Validate Kubernetes Manifests
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup Kustomize
+        run: |
+          curl -s "https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh" | bash
+          sudo mv kustomize /usr/local/bin/
+          kustomize version
+
+      - name: Validate Kustomize build
+        run: |
+          echo "Building kustomization..."
+          kustomize build deploy/kubernetes > /tmp/k8s-manifests.yaml
+          echo "Kustomize build successful!"
+          echo "Generated manifests:"
+          cat /tmp/k8s-manifests.yaml
+
+      - name: Setup kubeconform
+        run: |
+          wget https://github.com/yannh/kubeconform/releases/latest/download/kubeconform-linux-amd64.tar.gz
+          tar xf kubeconform-linux-amd64.tar.gz
+          sudo mv kubeconform /usr/local/bin/
+          kubeconform -v
+
+      - name: Validate manifests with kubeconform
+        run: |
+          echo "Validating Kubernetes manifests..."
+          kustomize build deploy/kubernetes | \
+            kubeconform -strict -summary \
+              -kubernetes-version 1.28.0 \
+              -schema-location default \
+              -schema-location 'https://raw.githubusercontent.com/datreeio/CRDs-catalog/main/{{.Group}}/{{.ResourceKind}}_{{.ResourceAPIVersion}}.json' \
+              -skip CustomResourceDefinition \
+              -ignore-missing-schemas
+
+      - name: Upload validated manifests
+        uses: actions/upload-artifact@v4
+        with:
+          name: k8s-manifests
+          path: /tmp/k8s-manifests.yaml
+          retention-days: 5
+
+  kind-integration-test:
+    name: kind Cluster Integration Test
+    runs-on: ubuntu-latest
+    needs: validate-manifests
+    timeout-minutes: 30
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Build semantic-router image
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          file: ./Dockerfile.extproc
+          tags: ghcr.io/vllm-project/semantic-router/extproc:test
```

Comment:
this could cause conflict between PRs, can you setup a local image registry and push the image there?